ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlans.py

```diff
@@ -44,9 +44,11 @@ def vlans(self) -> dict | None:
                     continue
 
                 if vlan_id in vlans:
-                    self._raise_duplicate_vlan_error(
-                        vlan_id, f"MLAG Peering VLAN in vrf '{vrf['name']}' (check for duplicate VRF VNI/ID)", tenant["name"], vlans[vlan_id]
-                    )
+                    if vlans[vlan_id]["name"] != f"MLAG_iBGP_{vrf['name']}":
+                        # Only raise if the duplicate is for a different VRF.
+                        self._raise_duplicate_vlan_error(
+                            vlan_id, f"MLAG Peering VLAN in vrf '{vrf['name']}' (check for duplicate VRF VNI/ID)", tenant["name"], vlans[vlan_id]
+                        )
```

Comment:
```suggestion
                if (vlan_id in vlans) and (vlans[vlan_id]["name"] != f"MLAG_iBGP_{vrf['name']}"):
                    # Only raise if the duplicate is for a different VRF.
                    self._raise_duplicate_vlan_error(
                        vlan_id, f"MLAG Peering VLAN in vrf '{vrf['name']}' (check for duplicate VRF VNI/ID)", tenant["name"], vlans[vlan_id]
                    )
```

You can merge both if's.